components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -56,6 +60,27 @@ CurlDownloadHandler::CurlDownloadHandler(
         m_http_headers.append("Cache-Control: no-cache");
         m_http_headers.append("Pragma: no-cache");
     }
+    std::set<std::string> const kReservedHeaders = {"range", "cache-control", "pragma"};
+    // RFC 7230 token pattern: one or more tchars
+    std::regex const header_name_pattern("^[!#$%&'*+.^_`|~0-9a-zA-Z-]+$");
+    // Must consist of printable ASCII characters (values between 0x20 and 0x7E)
+    std::regex const header_value_pattern("^[\\x20-\\x7E]*$");
+    for (auto const& [key, value] : custom_headers) {
```

Comment:
Personally I don't think it's necessary to check whether the given headers are valid. If invalid heads are given, curl will return errors and should be caught by the caller. @kirkrodrigues what do you think?

Reply:
Hm, how do you know curl will return errors for invalid headers? The [docs](https://curl.se/libcurl/c/CURLOPT_HTTPHEADER.html) say this:

> libcurl passes on the verbatim strings you give it, without any filter or other safe guards. That includes white space and control characters.

To me, that implies that something other than curl (e.g. the destination server) will indicate if the headers are invalid. That said, I think relying on something other than curl to do the error checking is fine (since header errors are an exceptional case) but we should at least validate this:

> The headers included in the linked list must not be CRLF-terminated, because libcurl adds CRLF after each header item itself.

Reply:
Sorry, my bad. What I mean is if the header contains an error, the HTTP request will fail and curl will return the error from the server, so we don't have to validate that. I think we're aligned with this, so the conclusion for Xiaochong is:
- We don't validate the header and leave this to the server
- We check the linked list is not CRLF-terminated as Kirk mentioned

Reply:
Are we still planning to remove the validation? @anlowee

Reply:
I think with these check are safer anyway, they also ensure that the header is not terminated with \r\n